commit 94781e961f810fd0374f1a6213b278a92fbfda68
Author: dmorgan <dmorgan@localhost.(none)>
Date:   Sun Jun 26 22:51:22 2011 +0200

    part of Backport from 4.2.1 provides becoming obsoletes bug (fpons)
    (is it still needed?)

diff --git a/lib/depends.c b/lib/depends.c
index 4daa512..2fe7a55 100644
--- a/lib/depends.c
+++ b/lib/depends.c
@@ -150,7 +150,7 @@ static void addObsoleteErasures(rpmts ts, tsMembers tsmem, rpm_color_t tscolor,
 	     * If no obsoletes version info is available, match all names.
 	     */
 	    if (rpmdsEVR(obsoletes) == NULL
-		     || rpmdsAnyMatchesDep(oh, obsoletes, _rpmds_nopromote)) {
+		     || rpmdsNVRMatchesDep(oh, obsoletes, _rpmds_nopromote)) {
 		char * ohNEVRA = headerGetAsString(oh, RPMTAG_NEVRA);
 		rpmlog(RPMLOG_DEBUG, "  Obsoletes: %s\t\terases %s\n",
 			rpmdsDNEVR(obsoletes)+2, ohNEVRA);
